src/alfasim_sdk/_internal/alfacase/case_description.py

```diff
@@ -2897,11 +2896,18 @@ def _check_fluid_references(self, *, reset_invalid_reference: bool = False):
         from itertools import chain
 
         elements_with_invalid_fluid = []
-        fluids_available = set(
+
+        # Update this tuple when a new PVT Model is added
+        known_pvt_models = ("compositional", "combined", "correlations", "tables")
+        all_fluids = [
             fluid
-            for compositional_model in self.pvt_models.compositional.values()
-            for fluid in compositional_model.fluids.keys()
-        )
+            for known_pvt_attr in known_pvt_models
+            for pvt_model in getattr(self.pvt_models, known_pvt_attr).values()
+            for fluid in (
+                pvt_model.fluids.keys() if hasattr(pvt_model, "fluids") else ()
+            )
+        ]
```

Comment:
I think we should extract this common code (also used in `get_duplicate_keys`) to a function, and reuse in both places, instead of copying/pasting the code.

Reply:
yup, good idea 👍